src/main/java/bio/terra/common/ValidationUtils.java

```diff
@@ -80,6 +81,30 @@ public static String requireNotBlank(String value, String errorMsg) {
     return value;
   }
 
+  public static LinkedHashMap<String, String> validateMatchingColumnDataTypes(
+      RelationshipTermModel fromTerm, RelationshipTermModel toTerm, List<TableModel> tables) {
+    LinkedHashMap<String, String> termErrors = new LinkedHashMap<>();
+    Optional<ColumnModel> fromColumn = retrieveColumnModelFromTerm(fromTerm, tables);
+    Optional<ColumnModel> toColumn = retrieveColumnModelFromTerm(toTerm, tables);
+    if (fromColumn.isPresent() && toColumn.isPresent()) {
+      TableDataType fromColumnDataType = fromColumn.get().getDatatype();
+      TableDataType toColumnDataType = toColumn.get().getDatatype();
+      if (!fromColumnDataType.equals(toColumnDataType)) {
```

Comment:
I think we may want this to be a little more lenient (e.g. text = string). Might be a bit of a pain to have to put together the list but I think we'll need it